pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdSources.java

```diff
@@ -315,6 +316,8 @@ protected void validateSourceConfigs(SourceConfig sourceConfig) {
         protected String DEPRECATED_sourceConfigString;
         @Parameter(names = "--source-config", description = "Source config key/values")
         protected String sourceConfigString;
+        @Parameter(names = "--batch-source-config", description = "Batch source config key/values")
+        protected String batchSourceConfigString;
```

Comment:
Curious to what an example key/values string looks like? Is it a json string?

Also, wondering if another flag call trigger-type should exist? At this point all the implementations are internal and in my opinion the user shouldn't need to know full class name just what type they want to use. The cli tool can take this information and add the correct classname.

Reply:
@cckellogg So you are suggesting that we would use a combination of the `--batch-source-config` flag to specify the cron expression and another flag called `batch-trigger-type` that would take two values, Immediate and Cron ?

Reply:
I think a `batch-trigger-type` (with cron and immediate as values) and then maybe a `batch-cron-expression`. We could do this later in another patch. I think we should add some documentation and examples of how to pass the `--batch-source-config` because it's not obvious to me and I would need to look at the source code.

Reply:
@cckellogg If there was ever a new trigger type added to the project or via a separate class by a user, then we would need to change this class to perform the mapping from the "simple" name to the actual class name. I think it is best to leave it as `--batch-source-config` to be consistent with the `--source-config`  switch that allows the user to send in all of the configs inside a single JSON object.

I will add an example of a valid JSON object to the documentation.

Reply:
sounds good